experimental/sdk/webpubsub-socketio-extension/src/EIO/components/web-pubsub-connection-manager.ts

```diff
@@ -0,0 +1,204 @@
+    );
+
+    this._webPubSubEventHandler = new WebPubSubEventHandler(this._webPubSubOptions.hub, {
+      path: this._webPubSubOptions.path,
+      handleConnect: async (req, res) => {
+        try {
+          var connectionId = req.context.connectionId;
+          debug(`onConnect, connectionId = ${connectionId}`);
+
+          var context = new ClientConnectionContext(this.serviceClient, connectionId, res);
+
+          /**
+           * Two conditions lead to returning reponse for connect event:
+           *   1. The connection is accepted or refused by EIO Server and the corresponding events are triggered.
+           *   2. Exception is thrown in following code
+           * As a defensive measure, a timeout is set to return response in 1000ms in case of both conditions don't happen.
+           */
+          var timeout = setTimeout( () => { if (!context.connectResponded) { res.fail(500, `EIO server cannot handle connect request with error: Timeout 1000ms`); }}, 1000);
+
+          var connectReq = this.getEioHandshakeRequest(req, context);
+
+          this._candidateSids.push(connectionId);
+          this._clientConnections.set(connectionId, context);
+          // @ts-ignore to access private `handshake` method
+          await this.linkedEioServer.handshake(WEBPUBSUB_TRANSPORT_NAME, connectReq);
+        } catch (error) {
+          debug(`onConnect, req = ${req}, err = ${error}`);
+          res.fail(500, `EIO server cannot handle connect request with error: ${error}`);
+          clearTimeout(timeout);
+        }
+      },
+
+      onConnected: async (req) => {},
+
+      handleUserEvent: async (req, res) => {
+        try{
+          var connectionId = req.context.connectionId;
+
+          debug(`onUserEvent, connectionId = ${connectionId}, req.data = ${req.data}`);
+
+          if (this._clientConnections.has(connectionId)) {
+            // @ts-ignore to access private `clients` property
+            const client = this.linkedEioServer.clients[connectionId];
+            
+            const packets = await client.transport.parser.decodePayload(req.data); // prettier-ignore
+
+            for (const packet of packets) {
+              client.onPacket(packet);
+            }
+            return res.success();
+          } else {
+            // `UserEventResponseHandler.fail(code, ...)` cannot set `code` with 404. Only 400, 401 and 500 are available.
+            return res.fail(400, `EIO server cannot find ConnectionId ${connectionId}`);
+          }
+        }
+        catch(err){
+          debug(`onUserEvent, req = ${req}, err = ${err}`);
+          return res.fail(500, `EIO server cannot handle user event with error: ${err}`);
+        }
+      },
+
+      onDisconnected: async (req) => {
+        var connectionId = req.context.connectionId;
+        debug(`onDisconnected, connectionId = ${connectionId}`);
+        if (!this._clientConnections.delete(connectionId)) {
```

Comment:
who cleans `linkedEioServer.clients`?

Reply:
It's a bug. it should be manually cleaned in `onDisconnected`. Updated